src/borg/locking.py

```diff
@@ -158,6 +190,47 @@ def is_locked(self):
     def by_me(self):
         return os.path.exists(self.unique_name)
 
+    def kill_stale_lock(self):
+        for name in os.listdir(self.path):
```

Comment:
This is as designed:

we loop around in the lock filenames doing stuff with them until we meet one that we cannot kill for whatever reason.
If we do meet one like that - there's no point in processing any further filenames so we just exit with False

We only return True if we completed the whole loop.